### PR TITLE
Support Vectara (https://vectara.com) as memory backend.

### PR DESCRIPTION
**Pull request by int64max**
Add Vectara (https://vectara.com) as memory backend.

Vectara is an end-to-end LLM-powered search platform. Compared to other memory backends in AutoGPT that use OpenAI Ada to generate embeddings, Vectara uses its own state of the art models to generate embeddings. This provides high relevance search results. Thus, Vectara is not simply a vector database, but an end-to-end neural search platform.

Vectara has a generous free tier (50mb text).

Disclaimer: I am a Vectara employee, and happy to maintain this memory backend. I'm also looking into personally contributing to other parts of Auto-GPT, as well as how Vectara as an org can contribute.

### CODE REVIEW

**Comment by ntindle** [comment]
Ping me in our discord

File: autogpt/memory/vectara.py
Comment (ntindle) [inline autogpt/memory/vectara.py]: how likely is this to change?

File: autogpt/memory/vectara.py
Comment (ntindle) [inline autogpt/memory/vectara.py]: can this be abstracted out of just vectara into a more useful global util for jwts?

File: autogpt/memory/vectara.py
Comment (ntindle) [inline autogpt/memory/vectara.py]: would love a bit more specific errors here. 403, 500, etc for common use cases so we don't get too many issues

File: autogpt/memory/vectara.py
Comment (ntindle) [inline autogpt/memory/vectara.py:88]: I'm assuming status existing is a bad thing? can you comment this if statement for why this is a negative path

**Review by ntindle — CHANGES_REQUESTED** [review]
First pass review for standards and QOL

Add Tests that cover your new backend please

File: autogpt/memory/vectara.py
Comment (int64max) [reply autogpt/memory/vectara.py]: Highly unlikely. All our customers use the token endpoint this way.
If we ever change this, we will have to coordinate with all existing customers.

File: autogpt/memory/vectara.py
Comment (int64max) [reply autogpt/memory/vectara.py]: Thanks for the suggestion. I've moved it to `utils.py`.

File: autogpt/memory/vectara.py
Comment (int64max) [reply autogpt/memory/vectara.py]: I've added more details to message error message here and elsewhere.

File: autogpt/memory/vectara.py
Comment (int64max) [reply autogpt/memory/vectara.py:88]: I've added an `_ok()` method, with a proper comment to describe the status field.

**Comment by int64max** [comment]
Thank you for the prompt review. I've addressed all your comments. Please take another look.

**Comment by int64max** [comment]
Can you kindly take a look at this PR? I tried joining the AutoGPT Discord channel, but it keeps saying "Accept Invite", followed by "Whoops, unable to accept invite". Thanks.

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by ntindle** [comment]
Hey, I've marked this as don't merge until the Memory Fixes are in. Sorry to keep it on hold longer, just no real way to test/validate functionality until the fixes are in

**Pull request closed without merging**
